### DISCUSSION TITLE
EFR32MG24 FOTA over BLE keeps disconnecting

### DISCUSSION THREAD

**Discussion by adri1mart1** [zephyrproject-rtos/zephyr#72466] [Q&A] [answered]
Hello zephyr community.

I have recently try to use the [xG24-DK2601B board](https://docs.zephyrproject.org/latest/boards/silabs/efr32xg24_dk2601b/doc/index.html) with the [SMP server sample](https://docs.zephyrproject.org/latest/samples/subsys/mgmt/mcumgr/smp_svr/README.html).

```
west build -b efr32xg24_dk2601b --sysbuild zephyr/samples/subsys/mgmt/mcumgr/smp_svr -- -DEXTRA_CONF_FILE="overlay-bt.conf"
```

At the moment I'm stuck when trying to upload the signed firmware over BLE, I'm using NRFConnect on Android and you can see attached a log that illustrate the disconnection. [Log_2024-05-08_13_48_01[1].txt](https://github.com/zephyrproject-rtos/zephyr/files/15248558/Log_2024-05-08_13_48_01.1.txt)
What's very hard to debug is I have no error on the NRF side and no error as well via UART on the dev kit. I can only see that I'm disconnecting systematically at the same time.

```
*** Booting MCUboot v2.1.0-rc1-1-gd4394c2f9b76 ***
*** Using Zephyr OS build v3.6.0-3544-g638f1d58dfcb ***
I: Starting bootloader
I: Primary image: magic=good, swap_type=0x2, copy_done=0x1, image_ok=0x1
I: Scratch: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3
I: Boot source: none
I: Image index: 0, Swap type: none
I: Bootloader chainload address offset: 0xc000
I: Jumping to the first image slot
[00:00:00.000,000] <inf> flash_gecko: Device flash-controller@50030000 initialized
*** Booting Zephyr OS build v3.6.0-3544-g638f1d58dfcb ***
[00:00:00.001,000] <inf> littlefs: LittleFS version 2.8, disk version 2.1
[00:00:00.001,000] <inf> littlefs: FS at flash-controller@50030000:0xfc000 is 66 0x2000-byte blocks with 512 cycle
[00:00:00.001,000] <inf> littlefs: sizes: rd 16 ; pr 16 ; ca 64 ; la 32
[00:00:00.013,000] <wrn> bt_hci_core: Controller to host flow control not supported
[00:00:00.014,000] <inf> bt_hci_core: Identity: B4:3A:31:E7:29:98 (public)
[00:00:00.014,000] <inf> bt_hci_core: HCI: version 5.3 (0x0c) revision 0x0000, manufacturer 0x02ff
[00:00:00.014,000] <inf> bt_hci_core: LMP: version 5.3 (0x0c) subver 0x0000
[00:00:00.015,000] <inf> smp_bt_sample: Advertising successfully started
[00:00:00.015,000] <inf> smp_sample: build time: May  8 2024 14:00:50
[00:00:22.943,000] <inf> smp_bt_sample: Connected
```

I'm using the latest version of zephyr, no change on the code but this flag `CONFIG_MCUMGR_GRP_OS_BOOTLOADER_INFO=y` that I have added because NRF logs were requesting this info and the not supported error popped up. 

There is a warning about controller host flow not supported but I have no idea if that's important or not. 

On a side note, I did the exact same operation (firmware update via BLE using NRF connect) with an NRF52832 dk successfully.

I have also tried the firmware upload using serial communication with the xG24-DK2601B board successfully. I had to add `zephyr,uart-mcumgr = &usart0;` in the dts and makes me wonder why no silicons labs have this set already ?

Any hint on how to investigate this problem would be very very helpful. 
Thanks, Adrien M.

**Comment by RomainPelletant** [comment]
Did you try to increase timeout or decrease packet size?
Did you find any workaround?

**Comment by adri1mart1** [comment, marked as answer]
Hello Romain,

I'm finally taking some time to give an answer. We finally successfully ran the FOTA over BLE using the EFR32MG24 dev kit. 

Using zephyr commit rev 5a8652b0c at that time (because zephyr 3.6.0 was not able to compile the SMP server example for this board and there was a big commit change about reorganizing SOC in March I think,zephyr 3.7.0 was not release when we worked on that task at that moment). 

We've made the following changes to the SMP server example prj.conf to make it work:

```
CONFIG_BT_PERIPHERAL_PREF_TIMEOUT=500
CONFIG_BT_L2CAP_TX_MTU=96
CONFIG_BT_BUF_ACL_RX_SIZE=100
CONFIG_BT_BUF_ACL_TX_SIZE=100
CONFIG_PSA_CRYPTO_ENABLE_ALL=y
CONFIG_MCUMGR_GRP_OS_BOOTLOADER_INFO=y
CONFIG_MCUMGR_TRANSPORT_BT_CONN_PARAM_CONTROL_TIMEOUT=500
```

Incoming plan is to use zephyr 3.7.0 and test this feature again. Probably we'll fine tune the timeout as well. 

I'm mark this discussion as closed.
